shaarli_client/utils.py

```diff
@@ -21,3 +22,27 @@ def generate_all_endpoints_parsers(subparsers, endpoints):
     """Generate all endpoints' subparsers from an endpoints dict"""
     for ep_name, ep_metadata in endpoints.items():
         generate_endpoint_parser(subparsers, ep_name, ep_metadata)
+
+
+def write_output(args, response):
+    """Write the API response to file or standard output"""
+    if not args.outfile:
+        if args.format == 'json':
+            print(response.json())
+        elif args.format == 'pprint':
+            print(json.dumps(response.json(), sort_keys=True, indent=4))
+        elif args.format == 'text':
+            print(response.text)
+    else:
+        try:
+            output_file = open(args.outfile, 'w+')
+        except:
+            raise IOError("File %s cannot be written" % args.outfile)
```

Comment:
`OSError` should be raised as we're supporting Python >= 3.4

From https://docs.python.org/3/library/exceptions.html#IOError :

> Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.